gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/AuthenticationFlowHandlerTest.java

```diff
@@ -965,17 +1148,19 @@ public void shouldRedirectToMFAChallengePage_adaptiveMFA_2() throws Exception {
     @Test
     public void shouldRedirectToMFAChallengePage_adaptiveMFA_2_user_auth() throws Exception {
         router.route().order(-1).handler(rc -> {
+            String challenge = "{#context.attributes['geoip']['country_iso_code'] == 'FR'}";
+            EnrollSettings enrollSettings = createEnrollSettings(true, false, MfaEnrollType.REQUIRED, 0, "");
+            ChallengeSettings challengeSettings = createChallengeSettings(true, MfaChallengeType.RISK_BASED, challenge);
```

Comment:
**todo** : based on the test case name, the test is related to adaptive MFA but the Challenge type is RISK_BASED, should be CONDITIONAL